app/plugin_install.go

```diff
@@ -110,6 +110,18 @@ func (a *App) installPluginLocally(pluginFile io.ReadSeeker, replace bool) (*mod
 		return nil, model.NewAppError("installPlugin", "app.plugin.invalid_id.app_error", map[string]interface{}{"Min": plugin.MinIdLength, "Max": plugin.MaxIdLength, "Regex": plugin.ValidIdRegex}, "", http.StatusBadRequest)
 	}
 
+	isCompatible, err := utils.CheckRequiredConfig(manifest.RequiredConfig, a.Config())
```

Comment:
I'm inclined to remove this check from the server altogether, and exclusively rely on the plugin to trigger this check in `OnActivate`. The nuance here is that plugins running on 5.15 or earlier would have to do this anyway, and those running on 5.16 or later wouldn't -- but it's never quite clear to a plugin author when they should stop doing it the old way.

It's probably simpler just to require a single path to obtain this behaviour. Another benefit would be that we're not stuck "supporting" it in the server forever, but could easily change the interface (for newly compiled plugins).